BASE/src/Microsoft.ApplicationInsights/Channel/InMemoryChannel.cs

```diff
@@ -186,6 +188,17 @@ public void Flush(TimeSpan timeout)
             }
         }
 
+        /// <summary>
+        /// Will send all the telemetry items stored in the memory asynchronously.
+        /// </summary>
+        /// <returns>The task to await. Task has true set on successful flush.</returns>
+        public async Task<bool> FlushAsync(CancellationToken cancellationToken)
+        {
+            var taskResult = Task.Run(() => this.Flush(default(TimeSpan)), cancellationToken);
+            await taskResult.ConfigureAwait(false);
+            return taskResult.IsCompleted;   
```

Comment:
I still don't understand this. Can you explain?
Note: with the `await` in the line above, is there any way to get here without `IsCompleted` being true?

Reply:
Incase of InMemoryChannel Flush. I want to just wrap existing Flush on a Task.Run. If Task is complete want to return success to caller.

return Task.Run(this.Flush(default(TimeSpan)), cancellationToken); ==> This link of code does not return bool or Task<bool>

Reply:
Your comment in another section will solve this issue. Creating a new Interface like ITelemetryChannel2 or IAsyncFlushable and not touching the InMemoryChannel at all. Creating Async in InMemeoryChannel will not create any difference.